docs: Update release process

ckadner opened the pull request:
### Proposed Changes

Elaborate the steps in the release process.

https://github.com/ckadner/modelmesh-serving/blob/update-release-process-doc/docs/release-process.md

### Updates since last review:

- [x] how to create a new `modelmesh-minio-examples` image 
  - [x] create a new repo for it
- [x] add a new section before the release
  - [x] update KServe Go dependency using the latest KServe release candidate and test it
  - [x] update the CRDs (`config/crd/bases`)
- [x] a new section on what happens after a release was published:
  - [x] verify the "check" results on last pushed tag
  - [x] update KServe
    - [x] website (https://github.com/kserve/website/pull/214)
    - [x] charts (https://github.com/kserve/kserve/pull/2645)
  - [x] work with Dan on a release blog

njhill commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
-1.  They simplify back porting critical bug fixes to a patch level release for a particular release stream (e.g., producing a `v0.6.1` from `release-0.6`), when appropriate.
+1. They allow a release wrangler or other developers to focus on a release without interrupting development on `main`,
+2. They allow developers to track the development of a release before a release candidate is declared,
+3. They simplify back porting critical bug fixes to a patch level release for a particular release stream (e.g., producing a `v0.6.1` from `release-0.6`), when appropriate.
```
nit: Any reason for changing the numbering here (same applies to below)?

ckadner replied:
you mean we should use 0.10 as an example or how we utilize Mardown numbering here?

njhill commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
-   - The `modelmesh-controller` image will be published via GitHub Actions.
-1. Upload generated install manifests to GitHub release assets.
-1. Be sure to create and push a new tag for the `kserve/modelmesh-minio-examples` image.
+It's generally a good idea to search the repo or `control`-`shift`-`f` for strings
```
Isn't find ctrl-F (or cmd-F on mac)?

ckadner replied:
I was trying to indicate a search over all files -- control-f (without shift) is a in-file only search

njhill commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
+
+2. Build and push a new tag for the `kserve/modelmesh-minio-examples` image. If
+   there have been no changes since the last release, you can create a new tag
+   from the previous release, e.g. creating a new `v0.10.0` tag from `latest`:
```
I guess we'll link to the minio build doc here once that's merged?

ckadner replied:
yes --we could leave this PR in draft for now until we have things settled -- or do a later update -- this instruction still works as is -- although not ideal

njhill commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
+11. Compare the release and release artifacts to those of previous releases to make
+    sure nothing was missed. Verify that the newly released version of the
+    [`modelmesh-controller`](https://hub.docker.com/r/kserve/modelmesh-controller/tags)
+    was pushed to DockerHub.
```
Should we mention anything about testing / checking fvt results?

ckadner replied:
yup -- my running list:

- FVT, test -- verify the "check" results on last pushed tag
- update KServe ... website, charts (more details to be gathered)
- work with Dan on a relase blog

njhill reviewed (commented):
Thanks @ckadner these look really good. Just had a couple of minor comments

rafvasq commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
+
+   The version tags should be updated in the following files:
+
+   - [ ] `config/default/config-defaults.yaml`: edit the image tags for ...
```
Nit: I like the helpful component checklist but could use "edit each component's image tags" to remove the `...`

rafvasq commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
+10. Create the new release in the GitHub UI from the `release-*` branch (or from the
+    tag created in the previous step). Enter the release tag value (i.e. `v0.10.0`) in
+    the "Release title" field and upload the generated install manifests ("Release assets")
+    in the "Atach binaries ..." section. Click the "Generate release notes" button which
```
Typo in "Attach"

rafvasq commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
-1.  They simplify back porting critical bug fixes to a patch level release for a particular release stream (e.g., producing a `v0.6.1` from `release-0.6`), when appropriate.
+1. They allow a release wrangler or other developers to focus on a release without interrupting development on `main`,
+2. They allow developers to track the development of a release before a release candidate is declared,
+3. They simplify back porting critical bug fixes to a patch level release for a particular release stream (e.g., producing a `v0.6.1` from `release-0.6`), when appropriate.
```
Nit: "backporting" instead of "back porting"

rafvasq commented on release/RELEASE_PROCESS.md:
```diff
@@ -2,48 +2,162 @@
+    in the "Atach binaries ..." section. Click the "Generate release notes" button which
+    will generate the release description.
+
+    **Note**, if you generated a pre-release (i.e. `v0.10.0-rc0`) then copy the release
```
Nit: "e.g." instead of "i.e."

rafvasq commented on release/RELEASE_PROCESS.md:
````diff
@@ -2,48 +2,162 @@
+   ```
+
+10. Create the new release in the GitHub UI from the `release-*` branch (or from the
+    tag created in the previous step). Enter the release tag value (i.e. `v0.10.0`) in
````
Nit: "e.g." instead of "i.e."

rafvasq reviewed (commented):
Just leaving a few proofreading-related comments! Thanks Christian.

ckadner commented:
Thanks for your review @rafvasq -- I addressed your comments and added the post-release section including the step about working with Dan on a release blog. If you have further guidance on that, please advice :-)

rafvasq commented on docs/release-process.md:
```diff
@@ -0,0 +1,249 @@
+5. Compare the release and release artifacts to those of previous releases to make
+   sure nothing was missed.
+
+6. Once the release as been published (a new tag has been pushed), verify the check
+   results by clicking on the check mark (✓) next to the latest commit on the `release-*`
+   branch.
```
Maybe it's a silly question, but does this check refer to something like [#308](https://github.com/kserve/modelmesh-serving/actions/runs/3898837488/workflow)? If so, where are FVTs verified if at all?

ckadner replied:
The FVTs are run on Pull Requests only for code changes (i.e. doc changes don't trigger it)

https://github.com/kserve/modelmesh-serving/blob/release-0.10/.github/workflows/run-fvt.yml#L4

So the last actual code change on the `release-*` branch would have made sure the code got tested.

rafvasq reviewed (commented):
Thanks Christian, just a few nits and a question about where FVTs fits in to all of this.

Re: the release blog and collaboration with Dan, I think you captured it fine. I ended up simply providing the ModelMesh portion of the blog to him [via a review](https://github.com/kserve/website/pull/227#discussion_r1098917277).

ckadner commented:
Thank you for your detailed review @rafvasq -- I pushed another update addressing your comments

rafvasq commented on docs/release-process.md:
```diff
@@ -0,0 +1,249 @@
+- [v0.7.0](https://kserve.github.io/website/blog/articles/2021-10-11-KServe-0.7-release)
+- [v0.8.0](https://kserve.github.io/website/0.8/blog/articles/2022-02-18-KServe-0.8-release/#modelmesh-updates)
+- [v0.9.0](https://kserve.github.io/website/0.9/blog/articles/2022-07-21-KServe-0.9-release/#inferenceservice-api-for-modelmesh)
+- [v0.10.0 draft](https://github.com/kserve/website/pull/227#discussion_r1098917277)
```
FYI, [v0.10's blog](https://kserve.github.io/website/0.10/blog/articles/2023-02-05-KServe-0.10-release/#modelmesh-updates) has now been released.

rafvasq reviewed (approved):
Thanks @ckadner, I left a link to v.10's blog but think there are enough examples and linking to the PR draft like you did is helpful too. Up to you if you want to include it too. Otherwise LGTM :)

njhill commented:
Thanks @ckadner!

njhill commented:
/lgtm
